app/models/spree/gateway/braintree_gateway_decorator.rb

```diff
@@ -45,10 +47,14 @@ def authorize money, credit_card, options={}
       # Include device data if present
       options[:device_data] = credit_card.device_data
 
+      current_merchant_account_id = BraintreePresenter.new(credit_card.payment_method).merchant_account_id(money)
+      options[:merchant_account_id] = current_merchant_account_id
+      credit_card.payment_method.preferences[:merchant_account_id] = current_merchant_account_id
```

Comment:
I think this line is not needed, and what it does is change the `default` value. 

You could also call the variable `merchant_account_for_transaction`, but I guess if you remove the third line, then you won't need the variable even